week-2/Homework/mandatory/4-sorting-algorithm.js

```diff
@@ -1,57 +1,87 @@
 /*
 At the start of the course, you worked in teams to sort your team members, labelled by
 numbers, in ascending or descending order.
-
 Today, you will be applying the sorting algorithm you used in that exercise in code! 
-
 Create a function called sortAges which:
 - takes an array of mixed data types as input
 - removes any non-number data types without using the built-in javascript filter method
-- returns an array of sorted ages in ascending order (without using the built-in javascript
-sort method 😎)
-
+- returns an array of sorted ages in ascending order 
+  - HARD MODE - without using the built-in javascript sort method ??
 You don't have to worry about making this algorithm work fast! The idea is to get you to
 "think" like a computer and practice your knowledge of basic JavaScript.
 */
+let newArr = [];
 
 function sortAges(arr) {
+    for (i = 0; i < arr.length; i++) {
+        if (!isNaN(arr[i])) {
+            newArr.push(arr[i])
+        }
+    }
 
+    for (i = 0; i < newArr.length; i++) {
+        for (j = 0; j < newArr.length; j++) {
+            if (newArr[j] > newArr[j + 1]) {
+                let term = newArr[j];
+                newArr[j] = newArr[j + 1];
+                newArr[j + 1] = term
+            }
+        }
+    }
+    return newArr
 }

```

Comment:
you have two small issues here (to allow you to pass test 2)
check where you put your variable newArr(search scope of variables) 
and search how isNaN works